src/ray/core_worker/core_worker.cc

```diff
@@ -613,24 +615,22 @@ void CoreWorker::Exit(
   /// otherwise the frontend code may not release its references and this worker will be
   /// leaked. See https://github.com/ray-project/ray/issues/19639.
   reference_counter_->ReleaseAllLocalReferences();
+
   // Callback to shutdown.
-  auto shutdown = [this, exit_type, creation_task_exception_pb_bytes]() {
-    // To avoid problems, make sure shutdown is always called from the same
-    // event loop each time.
-    task_execution_service_.post(
-        [this, exit_type, creation_task_exception_pb_bytes]() {
-          if (exit_type == rpc::WorkerExitType::CREATION_TASK_ERROR ||
-              exit_type == rpc::WorkerExitType::INTENDED_EXIT ||
-              exit_type == rpc::WorkerExitType::IDLE_EXIT) {
-            // Notify the raylet about this exit.
-            // Only CREATION_TASK_ERROR and INTENDED_EXIT needs to disconnect
-            // manually.
-            Disconnect(exit_type, creation_task_exception_pb_bytes);
-          }
-          Shutdown();
-        },
-        "CoreWorker.Shutdown");
-  };
+  auto shutdown =
+      [this, exit_type, detail = std::move(detail), creation_task_exception_pb_bytes]() {
+        // To avoid problems, make sure shutdown is always called from the same
+        // event loop each time.
+        task_execution_service_.post(
+            [this,
+             exit_type,
+             detail = std::move(detail),
```

Comment:
hmm, you can't move detail first and use it in line 629.

wondering if our *SAN caught this.
cc @mwtian

Reply:
I removed "move" from the above code, so I believe this part is okay to move?